jetbrains-core/src/software/aws/toolkits/jetbrains/core/region/AwsRegionProvider.kt

```diff
@@ -33,25 +40,24 @@ class AwsRegionProvider constructor(remoteResourceResolverProvider: RemoteResour
 
     override fun partitionData(): Map<String, PartitionData> = partitions
 
-    override fun defaultPartition(): AwsPartition = partitions()[DEFAULT_PARTITION]
-        ?: throw IllegalStateException("Could not find default partition: $DEFAULT_PARTITION")
-
-    override fun defaultRegion(): AwsRegion = try {
-        // Querying the instance metadata is expensive due to high timeouts and retries
-        val regionProviderChange = AwsRegionProviderChain(SystemSettingsRegionProvider(), AwsProfileRegionProvider())
-        regionProviderChange.region.id().let { regions(DEFAULT_PARTITION)[it] } ?: fallbackRegion()
-    } catch (e: Exception) {
-        LOG.warn(e) { "Failed to find default region" }
-        fallbackRegion()
+    override fun defaultPartition(): AwsPartition = defaultRegion().partitionId.let {
+        partitions()[it] ?: throw IllegalStateException("Could not find default partition: $it")
     }
 
-    private fun fallbackRegion(): AwsRegion = regions(DEFAULT_PARTITION).getOrElse(DEFAULT_REGION) {
-        throw IllegalStateException("Region provider data is missing default data")
+    override fun defaultRegion(): AwsRegion {
+        val regionIdFromChain = LOG.tryOrNull("Failed to find default region in chain", level = Level.WARN) {
+            regionChain.region.id()
+        }
+
+        val regionFromChain = regionIdFromChain?.let { regionId ->
+            this[regionId].onNull { LOG.warn { "Could not find $regionId in endpoint data" } }
```

Comment:
We do have `Logger.logWhenNull`

Reply:
But maybe this way reads better actually... No strong feelings

Reply:
oooh...